### PR TITLE
Add new document state: document-state-final

### PR DESCRIPTION
**Pull request by njohner**
With this PR we add a new document state, `document-state-final`. In this state the document cannot be modified anymore. 
- Only users with at least `LimitedAdmin` role and the user who finalized the document can reopen it. 
- A checked out document cannot be finalized
- A finalized document referenced in a pending task cannot be reopened
- A message is shown in the status bar of such documents to clearly state why the document cannot be modified.  
- We return the review state in the document status endpoint (needed by the frontend to update the document state and display a message in the document status bar)

For [CA-4433]

## Checklist
- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)
- Upgrade-Steps:
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
- API change: I did not update the API changelog as the `status` endpoint is not really an API endpoint and is not documented
  - [ ] Documentation is updated
  - [ ] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- New translations
  - [x] All msg-strings are unicode

[CA-4433]: https://4teamwork.atlassian.net/browse/CA-4433?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: opengever/document/tests/test_document_workflow.py [hunk 1]
```diff
@@ -87,32 +87,56 @@ def test_document_cannot_be_finalized_if_checked_out(self):
...
+        self.login(self.manager)
+        api.content.transition(obj=self.protected_document_with_task,
+                               transition=Document.finalize_transition)
+
+        with self.assertRaises(InvalidParameterError):
+            api.content.transition(
+                obj=self.protected_document_with_task,
+                transition=Document.reopen_transition)
+
+        api.content.transition(
```
Comment (phgross) [inline opengever/document/tests/test_document_workflow.py:113]: maybe `self.set_workflow_state('task-state-tested-and-closed', self.task_in_protected_dossier)` is easier to read.

File: opengever/document/locales/de/LC_MESSAGES/opengever.document.po [hunk 2]
```diff
@@ -672,15 +677,20 @@ msgid "label_yes"
...
 #. Default: "This item is being edited in Office Online by ${creator}."
-#: ./opengever/document/checkout/templates/checkedoutviewlet.pt
+#: ./opengever/document/checkout/templates/document_status_message_viewlet.pt
 msgid "message_collaborative_checkout_info"
 msgstr "Dieses Dokument wird von ${creator} gerade in Office Online bearbeitet."
 
+#. Default: "This document cannot be modified because it is final."
+#: ./opengever/document/checkout/templates/document_status_message_viewlet.pt
+msgid "message_final_info"
+msgstr "Ce document ne peut pas être modifié parce qu'il a été finalisé."
```
Comment (phgross) [inline opengever/document/locales/de/LC_MESSAGES/opengever.document.po:692]: Das Dokument wurde finalisiert und ist kann deshalb nicht mehr bearbeitet werden.

File: opengever/journal/handlers.py [hunk 3]
```diff
@@ -293,6 +294,19 @@ def reset_journal_history_after_clone(document, event):
     JournalManager(document).clear()
 
 
+DOCUMENT_STATE_CHANGED = 'Document state changed'
+
+
+def document_state_changed(context, event):
```
Comment (phgross) [inline opengever/journal/handlers.py:300]: Maybe add a small test for the journal, we had some issues recently in the journal.

Comment (phgross) [inline opengever/journal/handlers.py:300]: Does this event handler creates also a journal entry, when a document is initialized from the shadow state.

File: docs/public/user-manual/dokumente/finalisieren.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+.. _label-dokument-finalize:
+
+Dokumente finalisieren
+======================
+
+Ein Dokument kann mit der Aktion "Finalisieren" finalisert werden, so dass es nicht mehr bearbeitet werden kann. Der Benutzer der ein Dokument finalisiert und Administratoren können dann mit "Überarbeiten" das Dokument zurück in den Status Entwurf setzten, so dass es wieder bearbeite werden kann. Dieser Ablauf kann zum Beispiel verwendet werden um eine finale Version eines Dokument genehmigen lassen.
```
Comment (phgross) [inline docs/public/user-manual/dokumente/finalisieren.rst:6]: es wieder bearbeite**t** werden kann.

Comment (phgross) [reply docs/public/user-manual/dokumente/finalisieren.rst:6]: finale Version eines Dokument genehmigen **zu** lassen.

**Review by phgross — APPROVED** [review]
👍 LGTM - just some minor points.

File: opengever/journal/handlers.py (hunk 3 above)
Comment (njohner) [reply opengever/journal/handlers.py:300]: No because we skip the initialize transition, see below ⬇️

**Pull request merged**